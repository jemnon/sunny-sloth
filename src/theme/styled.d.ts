import 'styled-components';
import {
  Breakpoints,
  Colors,
  ColorMode,
  FocusColors,
  Fonts,
  FontSizes,
  Header,
  HoverColors,
  MaxWidths,
  Spacing,
  ZIndex,
} from '../types/theme';

declare module 'styled-components' {
  export interface DefaultTheme {
    breakpoints: Breakpoints;
    colors: Colors;
    focusColors: FocusColors;
    fonts: Fonts;
    fontSizes: FontSizes;
    header: Header;
    hoverColors: HoverColors;
    maxWidths: MaxWidths;
    spacing: Spacing;
    transition: string;
    zIndex: ZIndex;
  }
}
