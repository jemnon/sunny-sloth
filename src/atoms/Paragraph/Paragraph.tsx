import styled from 'styled-components';
import { up } from 'styled-breakpoints';
import { FontSizes } from '../../types/theme';

type FontStyle = 'normal' | 'italic';
type FontWeight = 'normal' | 'bold';
type FontSize = keyof FontSizes;

interface ParagraphProps {
  fontStyle?: FontStyle;
  fontSize?: FontSize;
  fontWeight?: FontWeight;
}

const Paragraph = styled.p<ParagraphProps>`
  font-family: ${({ theme }): string => theme.fonts.lato};
  font-weight: ${({ fontWeight = 'normal' }): string => fontWeight};
  font-size: ${({ theme, fontSize = 'f1' }): string =>
    theme.fontSizes[fontSize]};
  font-style: ${({ fontStyle = 'normal' }): string => fontStyle};
  ${up('sm')} {
    font-size: ${({ theme, fontSize = 'f2' }): string =>
      theme.fontSizes[fontSize]};
  }

  line-height: 1.5;

  & + &,
  & + * {
    margin-bottom: ${({ theme }): string => theme.spacing.sm4};
  }
  &:last-child {
    margin-bottom: 0;
  }
`;

export default Paragraph;
