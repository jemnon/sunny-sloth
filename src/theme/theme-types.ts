export interface Breakpoints {
  sm: string;
  md: string;
  lg: string;
  xlg: string;
}

export interface Fonts {
  lato: string;
  latoBold: string;
  latoItalic: string;
  noto: string;
  notoBold: string;
}

export interface Colors {
  cream: string;
  darkTeal: string;
  teal: string;
  lightBrown: string;
  nearBlack: string;
  nearWhite: string;
  darkOrange: string;
  orange: string;
  white: string;
  red: string;
  green: string;
}

export type ColorMode = 'dark' | 'light';

export interface MaxWidths {
  sm: string;
  md: string;
  lg: string;
}

export interface Spacing {
  sm1: string;
  sm2: string;
  sm3: string;
  sm4: string;
  md1: string;
  md2: string;
  md3: string;
  md4: string;
  lg1: string;
  lg2: string;
  lg3: string;
  lg4: string;
}

export interface ZIndex {
  z0: string;
  z1: string;
  z2: string;
  z3: string;
  z4: string;
  z5: string;
  z6: string;
  z999: string;
  z9999: string;
}
