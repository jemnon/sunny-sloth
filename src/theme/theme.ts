const theme = {
  breakpoints: {
    sm: '480px',
    md: '960px',
    lg: '1280px',
    xlg: '1440px',
  },
  fonts: {
    latoBold:
      `"Lato-Bold", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    lato:
      `"Lato", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    latoItalic:
      `"Lato-Italic", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    noto:
      `"Noto Serif TC", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    notoBold:
      `"Noto Serif TC-Bold", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
  },
  colors: {
    cream: '#f4ede6',
    darkTeal: '#162936',
    teal: '#244c53',
    lightBrown: '#d5a188',
    nearBlack: '#111',
    nearWhite: '#F9F9F9',
    darkOrange: '#BB5B34',
    orange: '#cf7651',
    white: '#ffffff',
    red: '#ff4136',
    green: '#19a974',
  },
  maxWidths: {
    sm: '480px',
    md: '960px',
    lg: '1280px',
  },
  spacing: {
    sm1: '0.25rem' /* 4px */,
    sm2: '0.5rem' /*  8px */,
    sm3: '0.75rem' /* 12px */,
    sm4: '1rem' /* 16px */,
    md1: '1.25rem' /* 20px */,
    md2: '1.5rem' /* 24px */,
    md3: '1.75rem' /* 28px */,
    md4: '2rem' /* 32px */,
    lg1: '2.25rem' /* 36px */,
    lg2: '2.5rem' /* 40px */,
    lg3: '2.75rem' /* 44px */,
    lg4: '3rem' /* 48px */,
  },
  transition: 'all 0.47s ease',
  zIndex: {
    z0: '0',
    z1: '1',
    z2: '2',
    z3: '3',
    z4: '4',
    z5: '5',
    z6: '6',
    z999: '999',
    z9999: '9999',
  },
};

export default theme;
