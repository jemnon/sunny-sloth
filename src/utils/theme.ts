import { DefaultTheme } from 'styled-components';

const theme: DefaultTheme = {
  breakpoints: {
    /* TODO: remove once conversion to styled-breakpoints is done */
    desktop: 'screen and (min-width: 960px)',
    tablet: 'screen and (min-width: 480px) and (max-width: 959px)',
    mobile: 'screen and (max-width: 479px)',
    /* ^^ */
    sm: '480px',
    md: '960px',
    lg: '1440px',
    xlg: '1920px',
  },
  fonts: {
    latoBold:
      `"Lato Bold", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    lato:
      `"Lato", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    latoItalic:
      `"Lato Italic", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    noto:
      `"Noto Serif TC", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
  },
  colors: {
    cream: '#f4ede6',
    darkTeal: '#162936',
    teal: '#244c53',
    lightBrown: '#d5a188',
    nearBlack: '#111',
    nearWhite: '#F9F9F9',
    darkOrange: '#BB5B34',
    orange: '#cf7651',
    white: '#ffffff',
    red: '#ff4136',
    green: '#19a974',
  },
  maxWidths: {
    sm: '480px',
    md: '960px',
    lg: '1440px',
  },
  spacing: {
    s1: '0.25rem' /* 4px */,
    s2: '0.5rem' /*  8px */,
    s3: '0.75rem' /* 12px */,
    s4: '1rem' /* 16px */,
    s5: '1.25rem' /* 20px */,
    s6: '1.5rem' /* 24px */,
    s7: '1.75rem' /* 28px */,
    s8: '2rem' /* 32px */,
    s9: '2.25rem' /* 36px */,
    s10: '2.5rem' /* 40px */,
    s11: '2.75rem' /* 44px */,
    s12: '3rem' /* 48px */,
  },
  transition: 'all 0.47s ease',
  zIndex: {
    z0: '0',
    z1: '1',
    z2: '2',
    z3: '3',
    z4: '4',
    z5: '5',
    z6: '6',
    z999: '999',
    z9999: '9999',
  },
};

export default theme;
