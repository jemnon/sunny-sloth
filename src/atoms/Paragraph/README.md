# Paragraph

### Description
Component description

```js
import Paragraph from '@fenderdigital/ui/FILL IN TYPEParagraph';
```

### Props
* `propName` - property name and description 
