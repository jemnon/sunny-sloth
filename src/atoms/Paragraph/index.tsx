import Paragraph, { ParagraphCSS } from './Paragraph';

export { ParagraphCSS };
export default Paragraph;
