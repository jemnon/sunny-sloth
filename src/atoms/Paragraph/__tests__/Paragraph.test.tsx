import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import Paragraph from '../Paragraph';
import theme from '../../../theme';

describe('<Paragraph />', (): void => {
  const component = renderer.create(
    <ThemeProvider theme={theme}>
      <Paragraph />
    </ThemeProvider>
  );
  it('exists', (): void => {
    expect(Paragraph).toBeDefined();
  });
  it('renders defaults correctly', (): void => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});