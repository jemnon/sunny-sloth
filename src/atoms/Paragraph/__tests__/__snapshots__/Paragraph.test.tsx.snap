// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Paragraph /> renders defaults correctly 1`] = `
<p
  className="sc-bdfBwQ ksIpcb"
>
  Paragraph
</p>
`;
